### PR TITLE
[Custom Amounts 4.2] Add Bottom sheet to select type of custom amount

### LINKED ISSUE #10246: Developing the 'Add Percentage' feature, which involves UI and business logic integration.

### PR DESCRIPTION
**Pull request by AnirudhBhat**
Closes: #10246 

### Description

This Pull Request introduces a bottom sheet that allows users to choose the method for creating custom amounts. There are two options available:

1. Fixed custom amount.
2. Percentage-based custom amount.

Note that if an order does not have any products or custom amounts added, the option to create a percentage-based custom amount will not be available. In such cases, the process will directly proceed to create a fixed custom amount, as the order must contain either a product or an existing custom amount to enable the percentage-based option.

#### Adding a new custom amount when the order total is zero
1. Navigate to the order creation screen
2. Click on the "Enter custom amount" button
3. Since there are no products or custom amounts added to the order, **ensure it takes you directly to the custom amount dialog without showing any bottom sheet**

#### Adding a new custom amount when the order total is NOT zero
1. Navigate to the order creation screen
2. Click on the "Enter custom amount" button
3. Since there are no products or custom amounts added to the order, it takes you directly to the custom amount dialog without showing any bottom sheet
4. Try to add another custom amount
5. Ensure it shows you a bottom sheet to select the type of custom amount to add
6. Ensure selecting appropriate option takes you to the appropriate dialog

#### Adding a percentage based custom amount
1. Navigate to the order creation screen
2. Click on the "Enter custom amount" button
3. Since there are no products or custom amounts added to the order, it takes you directly to the custom amount dialog without showing any bottom sheet
4. Try to add another custom amount
5. Ensure it shows you a bottom sheet to select the type of custom amount to add
6. Select percentage option
7. Play around with the screen and ensure proper custom amount is added according to the entered percentage

**Repeat the same for existing custom amounts by editing it and making sure all the details are correct**

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/1331230/757333d0-6bdc-40ba-9978-814675412736

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/CustomAmountTypeBottomSheetDialog.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import com.woocommerce.android.widgets.WCBottomSheetDialogFragment
+import javax.annotation.Nullable
+
+class CustomAmountTypeBottomSheetDialog : WCBottomSheetDialogFragment() {
+    private val sharedViewModel: OrderCreateEditViewModel by fixedHiltNavGraphViewModels(R.id.nav_graph_order_creations)
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        @Nullable container: ViewGroup?,
+        @Nullable savedInstanceState: Bundle?
+    ): View? {
```
Comment (rossanafmenezes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/CustomAmountTypeBottomSheetDialog.kt:22]: ❓ Just curious. Why are we setting the View as nullable here?

**Review by rossanafmenezes — APPROVED** [review]
LGTM

I left one question, just out of curiosity. 
Also, super nitpick,  I have noticed that if you select the percentage option, the EditText view is not in focus as it is when you select fixed amount:

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/CustomAmountTypeBottomSheetDialog.kt (hunk 1 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/CustomAmountTypeBottomSheetDialog.kt:22]: Good catch. @Nullable annotation is not required for this piece of code as the language's type system inherently distinguishes between nullable and non-nullable types. Fixed here: https://github.com/woocommerce/woocommerce-android/pull/10298/commits/918b3fcbe1953872f3c9e277b41ce64ed6c49c34

**Comment by AnirudhBhat** [comment]
> Also, super nitpick, I have noticed that if you select the percentage option, the EditText view is not in focus as it is when you select fixed amount:

Good catch. I've fixed this here: https://github.com/woocommerce/woocommerce-android/pull/10298/commits/d63c6e4d2e86dae98d6dd5185d3c19038cb80907

**Pull request merged**